lib/galaxy/authnz/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""
```

Comment:
Was there a reason to create this folder as separate from other auth stuff? Otherwise I think it would make more sense to have with the rest of the pluggable auth, under `galaxy/lib/galaxy/auth/providers/`, or even a `oauth2` subdirectory under providers.

Reply:
Though their logic might sound similar to some degree, but they have their own flow, constructors, managers, and backends (providers) which do not necessarily overlap with each other. Hence IMHO merging these two might be counter-intuitive.